lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -1,15 +1,207 @@
 $hostSource = @"
 using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Globalization;
+using System.Management.Automation;
 using System.Management.Automation.Host;
+using System.Security;
+using System.Text;
 using System.Threading;
 
-public class PuppetPSHost : PSHost
+namespace Puppet
 {
+  public class PowerShellRawUserInterface : PSHostRawUserInterface
+  {
+    public PowerShellRawUserInterface()
+    {
+      BufferSize = new Size(80, 25);
+      BackgroundColor = ConsoleColor.Black;
+      ForegroundColor = ConsoleColor.White;
+      CursorPosition = new Coordinates(0, 0);
+      CursorSize = 1;
+    }
+
+    public override ConsoleColor BackgroundColor { get; set; }
```

Comment:
Puppet.PowerShellRawUserInterface.BackgroundColor.get must declare a body because it is not marked as asbtract or extern